Network/DNS/Transport.hs

```diff
@@ -118,11 +117,15 @@ resolveOne ai gen q tm retry ctls rcv =
 -- UDP attempts must use the same ID and accept delayed answers
 -- but we use a fresh ID for each TCP lookup.
 --
+-- AXFR requests are only fulfilled over TCP.
+--
 udpTcpLookup :: IO Identifier -> UdpRslv
 udpTcpLookup gen retry rcv ai q tm ctls = do
     ident <- gen
-    udpLookup ident retry rcv ai q tm ctls `E.catch`
-            \TCPFallback -> tcpLookup gen ai q tm ctls
+    case qtype q of
+      AXFR -> tcpLookup gen ai q tm ctls
+      _    -> udpLookup ident retry rcv ai q tm ctls `E.catch`
+                 \TCPFallback -> tcpLookup gen ai q tm ctls
```

Comment:
Write a new low-level interface for AXFR, rather than changing the "lookup" interface.  AXFR is not a "lookup" for a single RRset.